synapse/storage/events.py

```diff
@@ -929,6 +929,10 @@ def _update_current_state_txn(self, txn, state_delta_by_room, max_stream_order):
                     txn, self.get_users_in_room, (room_id,)
                 )
 
+                self._invalidate_cache_and_stream(
```

Comment:
This doesn't seem to work in a synchrotron environment, presumably because the invalidation never gets called on the sync worker and so the sync worker keeps serving up the stale cache.  I'm not sure what the correct fix is though.  https://github.com/matrix-org/sytest/commit/b1dbc28aca5409617b3d93ee97364484d6b3e938 is a test for this, but it seems we don't run sytest against worker configs :|

Reply:
this is because i am a monumental crank and the master process wasn't running this branch - sowwy.